src/blog/2025-07-24-oxlint-type-aware.md

```diff
@@ -0,0 +1,164 @@
+---
+title: Oxlint Type-Aware Preview
+outline: deep
+authors:
+  - boshen
+  - camchenry
+  - auvred
+---
+
+<AppBlogPostHeader />
+
+<br>
+
+We're thrilled to announce type-aware rules support in Oxlint!
+
+This preview release aims to engage with the community for collaboration and
+discussion by documenting our decision process and technical details.
+
+## Quick Start
+
+If oxlint isn't already installed, please visit our [usage guide](https://oxc.rs/docs/guide/usage/linter.html).
+
+To get started, run:
+
+```bash
+pnpm add -D oxlint-tsgolint
+oxlint --tsconfig tsconfig.json
+```
+
+You'll then see two of our most requested type-aware rules in action:
+
+- no-floating-promises
+- no-misused-promises
+
+The next version will enable all type-aware rules.
+
+## Performance
+
+Our testing shows that large repositories, which previously took several minutes to run with `typescript-eslint`,
+now complete in less than 10 seconds.
+
+We achieved this by leveraging [`typescript-go`](https://github.com/microsoft/typescript-go),
+the [10x faster TypeScript](https://devblogs.microsoft.com/typescript/typescript-native-port) written in Go.
+
+## Type-Aware linting
+
+Please refer to
+[Rust-Based JavaScript Linters: Fast, But No Typed Linting Right Now](https://www.joshuakgoldberg.com/blog/rust-based-javascript-linters-fast-but-no-typed-linting-right-now)
+to understand the current status of type-aware linting in the ecosystem.
+
+## Technical Details
+
+The core of this new functionality is [oxc-project/tsgolint](https://github.com/oxc-project/tsgolint).
+
+The tsgolint project was initially prototyped as [typescript-eslint/tsgolint](https://github.com/typescript-eslint/tsgolint).
+However, the `typescript-eslint` team decided not to allocate development resources to this prototype, as they plan to continue their work on `typescript-eslint` for typed linting with ESLint.
```

Comment:
Worth pointing out (in a different chapter) what is the USP of using tsgolint with oxlint (instead of, e.g., waiting for typescript-eslint to be "go-powered")?